src/documents.rs

```diff
@@ -301,11 +302,36 @@ impl<'a> DocumentsQuery<'a> {
     }
 }
 
+#[derive(Debug, Clone, Serialize)]
+pub struct DocumentDeletionQuery<'a> {
+    #[serde(skip_serializing)]
+    pub index: &'a Index,
+
+    /// Filters to apply.
+    ///
+    /// Read the [dedicated guide](https://docs.meilisearch.com/reference/features/filtering.html) to learn the syntax.
+    pub filter: &'a str,
+}
+
+impl<'a> DocumentDeletionQuery<'a> {
+    pub fn new(index: &Index) -> DocumentDeletionQuery {
+        DocumentDeletionQuery { index, filter: "" }
+    }
+
+    pub fn with_filter<'b>(&'b mut self, filter: &'a str) -> &'b mut DocumentDeletionQuery<'a> {
+        self.filter = filter;
+        self
+    }
```

Comment:
If the filter is absolutely mandatory to make the request work we should get it from the `::new`, right?

```suggestion
    /// Filters to apply.
    ///
    /// Read the [dedicated guide](https://docs.meilisearch.com/reference/features/filtering.html) to learn the syntax.
    pub filter: &'a str,
}

impl<'a> DocumentDeletionQuery<'a> {
    pub fn new(index: &Index, filter: &str) -> DocumentDeletionQuery {
        DocumentDeletionQuery { index, filter }
    }
```

-----

Or maybe the filter should be optional, and we should let meilisearch returns the right error message, i.e.: « empty filter» vs « missing filter »

```suggestion
    /// Filters to apply.
    ///
    /// Read the [dedicated guide](https://docs.meilisearch.com/reference/features/filtering.html) to learn the syntax.
    pub filter: Option<&'a str>,
}

impl<'a> DocumentDeletionQuery<'a> {
    pub fn new(index: &Index) -> DocumentDeletionQuery {
        DocumentDeletionQuery { index, filter: None }
    }

    pub fn with_filter<'b>(&'b mut self, filter: &'a str) -> &'b mut DocumentDeletionQuery<'a> {
        self.filter = Some(filter);
        self
    }
```

Reply:
Let's go with the second option to take advantage of the errors returned by Meilisearch.